src/main/java/jcifs/smb1/netbios/NameServiceClient.java

```diff
@@ -0,0 +1,444 @@
+                            return;
+
+                        response.received = false;
+                        timeout -= System.currentTimeMillis() - start;
+                    }
+
+                } catch( InterruptedException ie ) {
+                    throw new IOException(ie.getMessage());
+                } finally {
+                    responseTable.remove( nid );
+                }
+
+                synchronized (LOCK) {
+                    if (NbtAddress.isWINS( request.addr ) == false)
+                        break;
+                                    /* Message was sent to WINS but
+                                     * failed to receive response.
+                                     * Try a different WINS server.
+                                     */
+                    if (request.addr == NbtAddress.getWINSAddress())
+                        NbtAddress.switchWINS();
+                    request.addr = NbtAddress.getWINSAddress();
+                }
+            }
+        }
+    }
+
+    NbtAddress[] getAllByName( Name name, InetAddress addr )
+                                            throws UnknownHostException {
+        int n;
+        NameQueryRequest request = new NameQueryRequest( name );
+        NameQueryResponse response = new NameQueryResponse();
+
+        request.addr = addr != null ? addr : NbtAddress.getWINSAddress();
+        request.isBroadcast = request.addr == null;
+
+        if( request.isBroadcast ) {
+            request.addr = baddr;
+            n = RETRY_COUNT;
+        } else {
+            request.isBroadcast = false;
+            n = 1;
+        }
+
+        do {
+            try {
+                send( request, response, RETRY_TIMEOUT );
+            } catch( IOException ioe ) {
+                if( log.level > 1 )
+                    ioe.printStackTrace( log );
+                throw new UnknownHostException( name.name );
+            }
+
+            if( response.received && response.resultCode == 0 ) {
+                return response.addrEntry;
+            }
+        } while( --n > 0 && request.isBroadcast );
+
+        throw new UnknownHostException( name.name );
+    }
+    NbtAddress getByName( Name name, InetAddress addr )
+                                            throws UnknownHostException {
+        int n;
+        NameQueryRequest request = new NameQueryRequest( name );
+        NameQueryResponse response = new NameQueryResponse();
+
+        if( addr != null ) { /* UniAddress calls always use this
+                              * because it specifies addr
+                              */
+            request.addr = addr; /* if addr ends with 255 flag it bcast */
+            request.isBroadcast = (addr.getAddress()[3] == (byte)0xFF);
+
+            n = RETRY_COUNT;
+            do {
+                try {
+                    send( request, response, RETRY_TIMEOUT );
+                } catch( IOException ioe ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)
Similar issue at line numbers 373.